### ISSUE TITLE
[crash] Something crashing very frequently. Intel Xe Graphics, Scaling 200%.

### ISSUE THREAD

**Issue by elevenhsoft** [pop-os/cosmic-epoch#262]
Hey.

Today installed Cosmic desktop because I want to rewrite my app from Iced to libcosmic. 
Unfortunately I'm getting crash very frequently. I don't know if it's compositor or panel or what. Maybe srceenshot will help
![IMG_20240416_220633](https://github.com/pop-os/cosmic-epoch/assets/114903054/1059f442-d761-4a15-871f-f2b5dd1353f5)

To back to work I'm just need to enter for example tty2 and back to tty1 and everything is back to normal for some time (few minutes). Apps etc are in place. 

When using gdm3 I was in lock screen after back to tty1 from tt2y so maybe it's locking my screen from time to time? 
For now I'm using cosmic-greeter because it's not giving me lock screen so changing tty fixes all the problems.

Please inform me where to find some logs from session so we can investigate those crashes.

I'm using Huawei Matebook X Pro with  11th Gen Intel i7-1165G7 and Intel Xe graphics (TigerLake-LP GT2). It's 3000x2000 screen with set scaling to 200%.

**Comment by Drakulix** [comment]
any chance you can provide some logs containing some of the crashes? e.g. run `journalctl --user _EXE=/usr/bin/cosmic-session > ~/cosmic.log` on tty2 should give you a log file in your home-directory.

**Comment by elevenhsoft** [comment]
Well, thank you. I dumped those logs. For me, there is nothing interesting so kind of weird. Looks like something else crashing?
[cosmic.log](https://github.com/pop-os/cosmic-epoch/files/15009690/cosmic.log)

It's log right after crash, log in to tty2 and dumped.

**Comment by elevenhsoft** [comment]
Next crash and again my log ends on this:

```
kwi 17 12:26:09 pop-os cosmic-session[1130]: 2024-04-17T10:26:09.136150Z  INFO Scale factor changed 2 for as surface in space "Panel" on eDP-1
kwi 17 12:51:33 pop-os cosmic-session[1130]: 2024-04-17T10:51:33.561311Z  INFO Scale factor changed 2 for as surface in space "Panel" on eDP-1
kwi 17 12:51:49 pop-os cosmic-session[1130]: 2024-04-17T10:51:49.572616Z  INFO Scale factor changed 2 for as surface in space "Panel" on eDP-1
```

**Comment by Drakulix** [comment]
Hmm, it looks like a bunch of components are having trouble reading some config files:
```
kwi 16 21:21:02 pop-os cosmic-session[1257]: 2024-04-16T19:21:02.516032Z ERROR stderr{client=InnerClientId { ptr: 0x5c5409170640, alive: true }}: com.system76.CosmicAppletPower:   ERROR  Error when loading Panel container config., err: RonSpanned(SpannedError { code: TrailingCharacters, position: Position { line: 1, col: 13 } })
kwi 16 21:21:02 pop-os cosmic-session[1257]: 2024-04-16T19:21:02.516039Z ERROR stderr{client=InnerClientId { ptr: 0x5c5409170640, alive: true }}: com.system76.CosmicAppletPower:     at cosmic-settings/src/pages/desktop/panel/inner.rs:62 on main
```

Not sure that is the issue, but could you try deleting `~/.config/cosmic` to reset your config state?

@wash2 Do you see possibly anything problematic in these log files? Looking at the screenshot and log we might create a lot of layer-surfaces for some reason. Possibly something isn't cleaning them up correctly in cosmic-comp, but I suspect the original issue is in some other component.

**Comment by elevenhsoft** [comment]
> Not sure that is the issue, but could you try deleting ~/.config/cosmic to reset your config state?
Just deleted whole cosmic folder. Crash happened about not even minute after reboot ^^
Sometimes screen flashing with this state what you can see on previous photo I attached.

**Comment by elevenhsoft** [comment]
Ok, got something else, this time it looks more interesting:

[cosmic-comp.log](https://github.com/pop-os/cosmic-epoch/files/15010193/cosmic-comp.log)

**Comment by Drakulix** [comment]
> Ok, got something else, this time it looks more interesting:
> 
> [cosmic-comp.log](https://github.com/pop-os/cosmic-epoch/files/15010193/cosmic-comp.log)

The log looks pretty normal, none of the messages are unusual. :S

**Comment by elevenhsoft** [comment]
Ok, so I won't give up ^^
Maybe this time: 
[cosmic-sudo.log](https://github.com/pop-os/cosmic-epoch/files/15010692/cosmic-sudo.log)

**Comment by elevenhsoft** [comment]
I removed com.system76.CosmicAppletPower from my panel and so far no crash at all. Seems stable. Maybe it's related?

**Comment by Drakulix** [comment]
Yes, there are a bunch of cosmic-applet-power related errors in your latest log.

**Comment by elevenhsoft** [comment]
I will observe this and give you a feedback after a while.

**Comment by wash2** [comment]
None of the Config errors should really cause a crash, because the values just fall back to defaults. Maybe you could try updating the submodules before building to see if the issue still occurs with the power applet enabled? I don't believe anything in the logs looks too weird, aside from the panel labeling logs for applets that arent the Power applet (the Time applet for example) as the Power applet

**Comment by elevenhsoft** [comment]
Removing this applet helps. Unfortunately some other like notifications (after clearing the notifications dialog) also crashing. However it's much  more stable in use without power applet.

**Comment by wash2** [comment]
> Removing this applet helps. Unfortunately some other like notifications (after clearing the notifications dialog) also crashing. However it's much more stable in use without power applet.

Does it always draw a bunch of panels when it crashes?

**Comment by elevenhsoft** [comment]
Opening and closing calendar produces some glitches too. Weird things mostly happening with applets.

**Comment by CosmicFusion** [comment]
Update: The issue is not Resolution related, when properly muxed to nvidia gpu 1920x1200 works

It's either and intel gpu issue

Or intel gpu + weird Resolution issue

**Comment by Drakulix** [comment]
Likely same issue as: https://github.com/pop-os/cosmic-comp/issues/499
